From: Parav Pandit <parav@nvidia.com>
Date: Wed, 27 Jan 2021 17:00:07 +0200
Subject: IB/mlx5: Return appropriate error code instead of ENOMEM
Patch-mainline: v5.12-rc1
Git-commit: d286ac1d05210695c312b9018b3aa7c2048e9aca
References: bsc#1103991 FATE#326007

When mlx5_ib_stage_init_init() fails, return the error code related to
failure instead of -ENOMEM.

Fixes: 16c1975f1032 ("IB/mlx5: Create profile infrastructure to add and remove stages")
Link: https://lore.kernel.org/r/20210127150010.1876121-8-leon@kernel.org
Signed-off-by: Parav Pandit <parav@nvidia.com>
Signed-off-by: Leon Romanovsky <leonro@nvidia.com>
Signed-off-by: Jason Gunthorpe <jgg@nvidia.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/hw/mlx5/main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/infiniband/hw/mlx5/main.c
+++ b/drivers/infiniband/hw/mlx5/main.c
@@ -5721,7 +5721,7 @@ err_mp:
 err_free_port:
 	kfree(dev->port);
 
-	return -ENOMEM;
+	return err;
 }
 
 static int mlx5_ib_stage_flow_db_init(struct mlx5_ib_dev *dev)
